app/models/post.rb

```diff
@@ -1,2 +1,7 @@
 class Post < ApplicationRecord
+  validates :title, presence: true, length: { minimum: 10, maximum: 50 }
+  validates :body, presence: true, length: { minimum: 10 }
+  validates :user_id, presence: true
+  belongs_to :user
+  has_many :comments
 end
```

Comment:
Since you have established associations between user and post, the post cannot be created unless a user is provided. So the validation in line 4 is not required. Please remove it.